catalyst/contrib/dl/callbacks/periodical_loader.py

```diff
@@ -0,0 +1,103 @@
+from typing import Mapping
+from collections import OrderedDict
+import copy
+
+from torch.utils.data import DataLoader
+
+from catalyst.core import Callback, CallbackOrder, State
+
+
+class PeriodicLoaderRunnerCallback(Callback):
+    """Callback for runing loaders with specified period.
+    To disable loader use ``0`` as period.
+
+    .. note::
+        ``'train'`` is a required loader and will
+        be ignored from passed loaders.
+
+    Example:
+
+        >>> PeriodicLoaderRunnerCallback(
+        >>>     train_additional=2,
+        >>>     valid=3,
+        >>>     valid_additional=5
+        >>> )
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Args:
+            kwargs: loader names and their run periods.
+
+                .. warning::
+                    Loder will be ignored if a period
+                    value is not a number or loader
+                    name is not presented in a list of loaders.
+        """
+        super().__init__(order=CallbackOrder.External)
+
+        self.valid_loader: str = None
+        self.valid_metrics: Mapping[str, float] = None
+        self.loaders: Mapping[str, DataLoader] = OrderedDict()
+
+        required_callbacks = {"train"}
+        self.loader_periods = {}
+        for loader, period in kwargs.items():
+            if not isinstance(loader, str) or not isinstance(
+                period, (int, float)
+            ):
+                continue
+            if loader in required_callbacks:
+                continue
+            self.loader_periods[loader] = int(period)
+
+    def on_stage_start(self, state: State) -> None:
+        """Collect information about loaders.
+
+        Arguments:
+            state (State): training state
+        """
+        # store pointers to data loader objects
+        for name, loader in state.loaders.items():
+            self.loaders[name] = loader
+        # stage validation loader
+        self.valid_loader = copy.copy(state.valid_loader)
+
+    def on_epoch_start(self, state: State) -> None:
+        """Set loaders for current epoch.
+
+        Arguments:
+            state (State): training state
+        """
+        epoch_num = state.epoch
+        # loaders to use in current epoch
+        epoch_loaders = OrderedDict()
+        for name, loader in self.loaders.items():
+            period = self.loader_periods.get(name, 1)
+            # ignore loaders where period - 0
+            if period > 0 and epoch_num % period == 0:
+                epoch_loaders[name] = loader
+        state.valid_loader = (
+            self.valid_loader
+            if self.valid_loader in epoch_loaders
+            else "train"
+        )
```

Comment:
sorry,  but we could not use `train` loader as `valid`  one;
I suggest to remember `valid_metrics` from `state` and pass them for the epochs without `valid_loader`

Reply:
oh, looks like it was required only as a trick around our requirement, that `valid_loader` in `loaders`....
in such case I suggest to use just the first key from loaders, rather than fix some name, like `train`

Reply:
Okay, I'll update that and will add information about this behaviour to docs.